### ISSUE TITLE
form generator copyAll not working for nonAdmins

### ISSUE THREAD

**Issue by rabauss** [contao/core-bundle#1450]
Unfortunately the `copyAll` action for forms of the form generator is not working for non admins in c4.4.

> "Not enough permissions to copyAll forms."

This is the default case of `tl_form->checkPermission()` - so `copyAll` is obviously not supported in form generator. 

Can we somehow hide the `copyAll`-Button of the global_operations or fix the `checkPermission()`?

**Comment by leofeyer** [comment]
How do I reproduce this?

**Comment by rabauss** [comment]
as an non admin (but with permission for forms create/delete) you have to go to `/contao?do=form&act=select` - then select some forms to copy and in the bottom form-button you should also be able to select the copyAll option. afterwards you'll see the mentioned error message!

I'm not sure if there is an option to hide the copyAll-action in the global-operations via `dca/tl_forms.php`.  Did I miss any dca-param for global_operations?

otherwise the switch case of `tl_form->checkPermission()` has to be extended for `copyAll` > here we need to make sure that permissions on the forms just copied are added to the current user! otherwise the current user will not see the copied forms!

**Comment by leofeyer** [comment]
To implement this, we need to add `copyAll` to the permission checks (somewhere [here](https://github.com/contao/core-bundle/blob/master/src/Resources/contao/dca/tl_form.php#L404)) and add the duplicated form IDs to the list of allowed forms as done [here](https://github.com/contao/core-bundle/blob/master/src/Resources/contao/dca/tl_form.php#L345-L389).

**Comment by leofeyer** [comment]
Sadly, the only fix I can offer at the time being is to remove the "copy all" button (see 7825f048925124e9ba2cc5769b1830c613abdbeb).

Non-admin users can only access the forms that have been enabled in the user settings. A few years ago, we have added a routine to allow non-admin users to copy forms, which are then automatically enabled in their settings. This works because Contao passes the ID of the newly generated form in the URL and we can access it in the `checkPermission()` method.

However, when copying multiple forms, the IDs are not getting passed and thus there is no way to automatically enable the new forms. So even though the forms would be generated, the user would not be able to see them.

We might be able to fix this in the future by enhancing the DC_Table or similar, but right now I do not have time to do this. Therefore I have added this as a known limitation.
